File: api/Portfolio.Domain/Models/Project.cs
using Portfolio.Domain.Models.Common;
using System.Collections.Generic;

namespace Portfolio.Domain.Models
{
    public class Project : BaseEntity, IHasDisplayNumber
    {
        #region Properties

        public string Title { get; set; }

        public string Description { get; set; }

        public string ImagePath { get; set; }

        public bool IsPublished { get; set; }

        public string DemoUrl { get; set; }

        public string GithubUrl { get; set; }

        public int DisplayNumber { get; set; }

        public ICollection<Skill> Skills { get; set; }

        #endregion
    }
}

File: public/src/app/data/Project.ts
import { BaseEntity } from './BaseEntity';
import { Skill } from './Skill';
import { Url } from './url';

export interface Project extends BaseEntity {
    title: string,
    description: string,
    imagePath: string,
    displayNumber: number,
    isPublished: boolean,
    skills? : Skill[],
    urls: Url[] | null
}
